src/hydro/srcterms/tabular_cooling.hpp

```diff
@@ -206,6 +236,35 @@ class TabularCooling {
 
   parthenon::Real EstimateTimeStep(parthenon::MeshData<parthenon::Real> *md) const;
 
+  // Get a lightweight object for computing cooling rate from the cooling table
+  const CoolingTableObj GetCoolingTableObj(parthenon::ParameterInput *pin) const {
+    using parthenon::Real;
+    Units units(pin);
+
+    // Create a lightweight object for computing cooling rates within kernels
+    const Real He_mass_fraction = pin->GetReal("hydro", "He_mass_fraction");
+    const Real mu = 1 / (He_mass_fraction * 3. / 4. + (1 - He_mass_fraction) * 2);
+    const Real mbar_over_kb = mu * units.mh() / units.k_boltzmann();
+    const Real adiabatic_index = pin->GetReal("hydro", "gamma");
+
+    return CoolingTableObj(log_lambdas_, log_temp_start_, log_temp_final_, d_log_temp_,
+                           n_temp_, mbar_over_kb, adiabatic_index, 1.0 - He_mass_fraction,
+                           units);
+  }
+  const CoolingTableObj
+  GetCoolingTableObj(std::shared_ptr<parthenon::StateDescriptor> hydro_pkg) const {
+    using parthenon::Real;
+    // Create a lightweight object for computing cooling rates within kernels
+    const auto mbar_over_kb = hydro_pkg->Param<Real>("mbar_over_kb");
+    const auto adiabatic_index = hydro_pkg->Param<Real>("AdiabaticIndex");
+    const auto He_mass_fraction = hydro_pkg->Param<Real>("He_mass_fraction");
+    const auto units = hydro_pkg->Param<Units>("units");
+
+    return CoolingTableObj(log_lambdas_, log_temp_start_, log_temp_final_, d_log_temp_,
+                           n_temp_, mbar_over_kb, adiabatic_index, 1.0 - He_mass_fraction,
+                           units);
+  }
+
```

Comment:
I was hoping to just go with the `StateDescriptor` one  in order to use the already computed `mbar_over_kb`. Then I found that where the cooling table is tested in a debugging function the `StateDescriptor` is not immediately available, so I added the `ParameterInput` one.

I think the better solution is to make it a member variable of `TabularCooling` but also pass in `hydro_pkg` to the constructor of `TabularCooling` just so that the computation of `mbar_over_kb` is centralized.